contracts/TrueReward/IEarnFinancialOpportunity.sol

```diff
@@ -1,4 +1,84 @@
 pragma solidity ^0.5.13;
 
-contract IEarnFinancialOpportunity {
+import "../TrueCurrencies/TrueRewardBackedToken.sol";
+import "./yTrueUSDInterface.sol";
+import "../TrueCurrencies/Proxy/OwnedUpgradeabilityProxy.sol";
+import "../TrueCurrencies/IFinancialOpportunity.sol";
+import "./IdGenerator.sol";
+
+contract IEarnFinancialOpportunity is IFinancialOpportunity {
+    using SafeMath for uint256;
+
+    yTrueUSDInterface public yToken;
+    TrueRewardBackedToken public token;
+    mapping (address => uint256) public yTokenBalance;
+
+    modifier onlyOwner() {
+        require(msg.sender == proxyOwner(), "only owner");
+        _;
+    }
+
+    modifier onlyToken() {
+        require(msg.sender == address(token), "only token");
+        _;
+    }
+
+    function configure(
+        yTrueUSDInterface _yToken,
+        TrueRewardBackedToken _token
+    ) public onlyOwner {
+        yToken = _yToken;
+        token = _token;
+    }
+
+    function proxyOwner() public view returns(address) {
+        return OwnedUpgradeabilityProxy(address(this)).proxyOwner();
```

Comment:
Also, you shouldn't staticcall into yourself if you don't have to; it's gas-intensive